Automattic/wp-calypso
# Add DIFM Price breakdown in page picker page

#### Proposed Changes
Adds a dummy basket to the page picker step so that the customer has some feedback of the cost incurred when purchasing extra pages.

#### Implementation Details
- New site 
  - [x] Show a dummy basket with a premium plan, the DIFM product with a price breakdown when available
  
- Existing site
  - [x] Existing site will add items to the calypso cart and then show the items with the exact dummy UI design same as above
  - [x] Existing site which has a paid plan will not show the pro plan in the basket (done in the backend)
  - [x] Existing site which does not have a paid plan will show the premium plan in the basket basket
  
- Common
  - [x] Implementation is mobile ready
  - [x] The DIFM breakdown and the total will be updated as the relevant number of extra pages area selected
  - [x] Hide all changes behind the `difm/allow-extra-pages` feature flag 
  
---

### Without Extra Pages
| Mobile | Desktop |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/3422709/189655477-d6dde2fc-53f8-44fb-8c49-36ad2f4836f9.png) | <img width="1433" alt="image" src="https://user-images.githubusercontent.com/3422709/189611974-d9e4889b-ef1c-453b-b0c0-f6ee80da0747.png"> |

### With Extra Pages
| Mobile | Desktop |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/3422709/189654480-ac0253ef-dbb8-467e-ae00-ebff911e4105.png) | ![image](https://user-images.githubusercontent.com/3422709/189612628-baab1003-80af-4e54-8e01-9cb55e7ed48a.png) |


#### Testing Instructions
- Apply DIFF D87491-code
- Sandbox public API
- Go through the flow `/start/do-it-for-me`
- On the page picker step apply the feature flag and make sure the extra page features are visible
     - `/start/do-it-for-me/difm-page-picker?flags=no-force-sympathy,difm/allow-extra-pages`
- Remove the feature flag and make sure no changes are visible to the page when comparing with production
    - `/start/do-it-for-me/difm-page-picker?flags=no-force-sympathy,-difm/allow-extra-pages`
- Make sure *Implementation Details* are working properly

#### Pre-merge Checklist






Related to #
